sdlib/d/gc/extent.d

```diff
@@ -400,6 +470,32 @@ unittest finalizers {
 	assert(cast(void*) largePd.extent.finalizer == cast(void*) &destruct);
 	assert(largePd.extent.usedCapacity == 19999);
 	threadCache.free(large);
+
+	// Basic test for small allocs:
+	auto small = threadCache.allocAppendable(1000, false);
+	auto smallPd = threadCache.getPageDescriptor(small);
+
+	import d.gc.slab;
+	auto sg = SlabAllocGeometry(smallPd, small);
+	auto idx = sg.index;
+	auto e = smallPd.extent;
+	assert(e.getTotalCapacity(idx) == 1024);
+
+	// Set a finalizer:
+	e.setFinalizer(idx, &destruct);
+	assert(e.hasFinalizer(idx));
+	assert(e.getTotalCapacity(idx) == 1016);
+
+	foreach (ushort i; 0 .. 1017) {
+		// Confirm that setting freespace does not clobber finalizer:
+		e.setFreeSpace(idx, i);
+		assert(cast(void*) e.getFinalizer(idx) == cast(void*) &destruct);
+		// Confirm that setting finalizer does not clobber freespace:
+		e.setFinalizer(idx, &destruct);
+		assert(e.getFreeSpace(idx) == i);
+	}
+
+	threadCache.free(small);
```

Comment:
It doesn't look like you are testing that you get the proper finalizer back. You probably want to have two and oscillate between them.